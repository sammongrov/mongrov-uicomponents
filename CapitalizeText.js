import React from 'react';
import { Text } from 'react-native';
import PropTypes from 'prop-types';

const CapitalizeText = (props) => {
  const { children } = props;
  const text = children
    ? children.slice(0, 1).toUpperCase() + children.slice(1, children.length)
    : '';
  return <Text {...props}>{text}</Text>;
};

CapitalizeText.propTypes = {
  children: PropTypes.string,
};

CapitalizeText.defaultProps = {
  children: '',
};

export default CapitalizeText;
