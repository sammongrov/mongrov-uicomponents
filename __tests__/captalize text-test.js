import React from 'react';
import renderer from 'react-test-renderer';
import CapitalizeText from '../CapitalizeText';

it('Badge renders correctly', () => {
  const tree = renderer.create(<CapitalizeText text="children" />).toJSON();
  expect(tree).toMatchSnapshot();
});
