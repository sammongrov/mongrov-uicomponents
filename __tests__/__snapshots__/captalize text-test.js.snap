// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  text="children"
>
  
</Text>
`;
